### ISSUE TITLE
Drawing Vector Point Off Board Bounds

### ISSUE THREAD

**Issue by swordmaster2k** [rpgtoolkit/editor#15]
Maximise a board of suitable size, one smaller than the maximised window. Using the vector tool start drawing a vector from the blank space in the window and it should add a point. 

This behaviour is incorrect, need to catch any attempt to route vector or program points outside of the boards dimensions.

**Comment by goblinJoel** [comment]
Are you sure you want to disallow this? If I understand the example, going off-board would be easier than clicking on the exact edge of the board when attempting to make a vector that extends to the edge. If you do disallow vector points outside the board bounds, I recommend snapping to the board edges (probably the point where the over-extended line crosses the edge) instead of canceling the point.

**Comment by swordmaster2k** [comment]
Good point, that's probably the way it will be handled if "snap to grid" is enabled. Really what I'm going to stop is users adding points outside the bounds of the board altogether. I'll consider drawing to the edge with "snap to grid" as a special case, otherwise I could simply insert the point where the vector cuts the board's axis

**Comment by chutchinson** [comment]
I disagree that the behavior is incorrect. Ideally for the board editor I imagine Adobe Photoshop. You should be able to pan, zoom, and otherwise navigate the board. There would be a border outside the board dimensions. You should be able to create vector points outside of the board dimensions. It may be easier to encapsulate a structure that way, a trigger zone, path, etc. This is a content creation tool, I would argue restricting user creativity or freedom is disingenuous.. The behavior you mentioned for **snap to grid** is reasonable and smart, and I agree with that completely.

**Comment by swordmaster2k** [comment]
I'll take a look at other editors and considered what's best for the board editor. Paint allows me to draw lines to points beyond its canvas. Snap to grid will probably be a special case where this can't happen because the bounds of the grid end before the free space.

I'm not sure how the current engine handles vectors with points defined outside the bounds of the board. Does it consider them valid or chuck an error out? I'll investigate. Another issue with this is users could create vectors in free space by accident and not see them, thus they could potentially create unwanted vectors.

Perhaps I'll disable starting vectors in free space at least, paint doesn't allow me to start lines in free space.

**Comment by goblinJoel** [comment]
A vector entirely outside the bounds of the board may never get triggered (unless you can programmatically move it at runtime), but I don't see why it would hurt anything unless the engine programmers are doing something strange.

Sounds like the board editor is going to be pretty neat!

**Comment by chutchinson** [comment]
I think you should be able to do almost anything to game assets at runtime, and moving a board vector was exactly the example I was going to use. It's not difficult to imagine drawing a board vector out of bounds, then move it in for a cutscene, minigame, etc.

**Comment by goblinJoel** [comment]
Will there be a listing of vectors or other objects in the board editor? I think the freedom to place things out of bounds (for whatever reason, temporary or not) would be useful for some developers, but this would necessitate some kind of way to select vectors whether they're visible in the window or not. Does the editor allow zooming?

**Comment by swordmaster2k** [comment]
There is not listing of vectors, its single selection with right mouse click. So drawing vectors off board makes them impossible to select at the moment. Perhaps you could provide the ability to dynamically create vectors at run time as an alternative? Which would mean they would only come into existence when they are actually needed.

Zooming? Sure, along with a toggled grid, coordinates, multitile selection, pencil, bucket tool, drawing of vectors, programs, sprites. To be honest the board editor could be improved no end, but for any initial release I'm going to focus on core functionality. Otherwise it would go on for ever.
